### PR TITLE
MAT-7355 added audit logs for associating cms id

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-7355](https://jira.cms.gov/browse/MAT-7355)
(Optional) Related Tickets:

### Summary
added audit logs for associating cms id

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/ActionLogService.java [hunk 1]
```diff
@@ -18,7 +19,11 @@ public class ActionLogService {
   private final MeasureActionLogRepository actionLogRepository;
 
   public boolean logAction(
-      final String targetId, Class targetClass, final ActionType actionType, final String userId) {
+      final String targetId,
+      Class targetClass,
+      final ActionType actionType,
+      final String userId,
+      final String... additionalActionMessage) {
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/ActionLogService.java:26]: not sure about these 3 dots

File: src/main/java/cms/gov/madie/measure/services/MeasureService.java [hunk 2]
```diff
@@ -685,6 +685,23 @@ public MeasureSet associateCmsId(
         qdmMeasureId,
         measureSet.getCmsId());
 
+    String associationSuccessMessage =
+        "Measures with IDs %s, %s are Associated with CMS ID %s on %s.";
+    String copyMetaDataStatusMessage =
+        copyMetaData ? "Metadata was copied over" : "Metadata was NOT copied over";
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/MeasureService.java]:
```suggestion
    String copyMetaDataStatusMessage =
        copyMetaData ? " Metadata was copied over" : " Metadata was NOT copied over";
```

File: src/main/java/cms/gov/madie/measure/services/MeasureService.java [hunk 3]
```diff
@@ -685,6 +685,23 @@ public MeasureSet associateCmsId(
         qdmMeasureId,
         measureSet.getCmsId());
 
+    String associationSuccessMessage =
+        "Measures with IDs %s, %s are Associated with CMS ID %s on %s.";
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/MeasureService.java]: can we make it lil bit more clear about QDM and QICore measures like which id is QDM and which one is QICore

File: src/main/java/cms/gov/madie/measure/services/ActionLogService.java (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/services/ActionLogService.java:26]: I was using these dots to make it optional

File: src/main/java/cms/gov/madie/measure/services/MeasureService.java (hunk 3 above)
Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/services/MeasureService.java]: Sure I can make the change

File: src/main/java/cms/gov/madie/measure/services/MeasureService.java (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/services/MeasureService.java]: Sure I can make the change

File: src/main/java/cms/gov/madie/measure/services/ActionLogService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/ActionLogService.java:26]: ah..okay, thanks!

**Review by adongare — APPROVED** [review]

**Pull request merged**